tests/unit/data-table/DataTable.spec.js

```diff
@@ -0,0 +1,204 @@
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: 'device 1' },
+                    { value: 'device 3' },
+                    { value: 'device 21' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['device 21', 'device 3', 'device 1'])
+        })
+
+        it('Treats null and undefined as equivalent to empty string', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    {},
+                    { value: 'zebra' },
+                    { value: null },
+                    { value: 'apple' },
+                    { value: '' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([undefined, null, '', 'apple', 'zebra'])
+        })
+
+        it('Sorts booleans descending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: true },
+                    { value: false },
+                    { value: true }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([false, true, true])
+        })
+
+        it('Groups booleans together by treating them as strings', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: true },
+                    { value: false },
+                    { value: true },
+                    { value: 'a-string' },
+                    { value: 'z-string' },
+                    { value: 2 },
+                    { value: 1 },
+                    { value: 0 }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([true, true, false, 0, 1, 2, 'a-string', 'z-string'])
```

Comment:
Food for thought only @Pezmc - I am happy to be told otherwise - this is just my interpretation. Please canvas others opinions before accepting my suggestions.

<br>

---


<br>

`[true, false, true, 'a-string', 'z-string', 2, 0, 1]`  becomes ==> `[true, true, false, 0, 1, 2, 'a-string', 'z-string']` - This feels a slightly weird to me. 

Since table sorting is for textual display (i.e. everything is a string to the end user) should the results not be...

```
[ 0, 1, 2, 'a-string', false, true, true, 'z-string' ]
```
Please node:  I dont mean to say numbers should be sorted as string for example I would expect that `[true, 55, false, 0, 1, 2, 'a-string', 'z-string']` would become `[ 0, 1, 2, 55, 'a-string', false, true, 'z-string' ]`


<br>

Alternatively, if considering `false` to be akin to `0` and `true` akin to `1` then sorting `asc` might want to result in 

```
[ 0, false, 1, true, true, 2, 'a-string', 'z-string' ]
```

Reply:
> Since table sorting is for textual display (i.e. everything is a string to the end user) should the results not be...

This isn't strictly true, since data table currently render true/false as check icon and blank (sidenote: should this show a cross for false), which feels strange when they're sorted alphabetically as a string.

<img width="941" alt="Screenshot 2022-10-24 at 12 41 30" src="https://user-images.githubusercontent.com/507155/197508286-b97d5269-dd92-4e04-824b-8a11aca73510.png">

#### Sorted as a group (this PR)
<img width="423" alt="Screenshot 2022-10-24 at 12 46 03" src="https://user-images.githubusercontent.com/507155/197509221-7df3eb02-0853-4b02-884c-ac58f9df1922.png">

#### Booleans treated as strings alphabetically
<img width="417" alt="Screenshot 2022-10-24 at 12 47 06" src="https://user-images.githubusercontent.com/507155/197509201-2c279ea5-6cb2-403b-9ee7-97ffb28158ad.png">

Reply:
Ah, of course, check box! Makes sense. 

1 thing: I guess these demo screen-caps include commit [1bbf300 Treat false as less than true](https://github.com/flowforge/forge-ui-components/pull/80/commits/1bbf300b66cbf6bc4d617de0ee94dc53594dd670) ?

In other words, my initial statement of 

> `[true, false, true, 'a-string', 'z-string', 2, 0, 1]` becomes ==> `[true, true, false, 0, 1, 2, 'a-string', 'z-string']` - This feels a slightly weird to me.

is no longer valid since `[true, false, true, 'a-string', 'z-string', 2, 0, 1]` would now become ==> `[false, true, true, 0, 1, 2, 'a-string', 'z-string']` due to 1bbf300?   Is that a fair stattement?

Reply:
Yes, the screencraps are after that commit! Though I do wonder if it makes more sense "visually" for true to come first (as it was before that commit), even though locally/programatically it should be last.

Reply:
> (sidenote: should this show a cross for false), which feels strange when they're sorted alphabetically as a string.

Agreed.  Either an `X` or the check box outline should be shown e.g. 
![image](https://user-images.githubusercontent.com/44235289/197510911-9125be64-c242-487e-b72b-fae16e728fd2.png)

One for the issues?